IdentityCore/src/throttling/cache/MSIDThrottlingCacheAccessor.h

```diff
@@ -0,0 +1,55 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+@class MSIDThrottlingCacheRecord;
+
+//singleton instances of MSIDServerDelayCache & MSIDUIRequiredCache will be in the implementation file as private properties.
+//Additionally, there will be dummy head and tail node in the .m file to maintain pseudo doubly-linked-list so we can better handle all corner cases.
+@interface MSIDThrottlingCacheAccessor : NSObject
+
+//Will implement getters in the .m file. 
+@property (nonatomic, readonly) NSDate *lastCleanUpTimeForUICache;
+@property (nonatomic, readonly) NSDate *lastCleanUpTimeForServerDelayCache;
+
+- (instancetype)initializeThrottlingCacheAccessor;
```

Comment:
if there're no parameters, there's no need to declare a custom initializer. You can just override -(instancetype)init

Reply:
yeah. I don't need explicit initializer for the cache accessor and the two underlying cache objects.
They simply provide CRUD APIs that will modify the contents of static MSIDCache objects that reside in the cache objects' .m files, just like how we implemented device assertion header & PRT header in the browser SSO PRT controller.